proto/substrait/algebra.proto

```diff
@@ -631,6 +631,10 @@ message WriteRel {
   OutputMode output = 6;
   RelCommon common = 7;
 
+  // The relation that is used with UPDATE to determine the new values of the fields.
+  // The schema must match with table_schema
+  ProjectRel update = 9;
```

Comment:
Given that a project relation could have as its input an arbitrary relation we could just make this a relation and suggest that it be a project relation.

But do we want the values coming from another plan subtree?  It may be better to have the expressions work over the same input already provided to the WriteRel.

Reply:
Yes, having the update expressions directly here would achieve that goal.

Reply:
fixed it will raise a PR on the public repo.